File: Website/wwwroot/scripts/components/consignment-search-form.ts
import 'app/appPage';

export default class ConsignmentSearchForm {
    public static run(): void {
        var totalGrossWeightMin = $("#TotalGrossWeightMin");
        var totalGrossWeightMax = $("#TotalGrossWeightMax")
        var totalValueMin = $("#TotalValueMin");
        var totalValueMax = $("#TotalValueMax");
        var totalPackagesMin = $("#TotalPackagesMin");
        var totalPackagesMax = $("#TotalPackagesMax")
        var totalNetWeightMin = $("#TotalNetWeightMin");
        var totalNetWeightMax = $("#TotalNetWeightMax")

        var date = $("#Date");
        var dateMax = $("#DateMax");
        var expectedDate = $("#ExpectedDate");
        var expectedDateMax = $("#ExpectedDateMax");

        var expectedDateDep = $("#ExpectedDateOfDeparture");
        var expectedDateMaxDep = $("#ExpectedDateOfDepartureMax");

        var dateCons = $("#DateCons");
        var dateMaxCons = $("#DateMaxCons");
        var expectedDateCons = $("#ExpectedDateCons");
        var expectedDateMaxCons = $("#ExpectedDateMaxCons");



        totalGrossWeightMin.change(function () {
            if (parseInt(totalGrossWeightMax.val()) < parseInt(totalGrossWeightMin.val())) {
                alert("Max value must be greater then min value");
                totalGrossWeightMin.val("");
            }
        });

        totalGrossWeightMax.change(function () {
            if (parseInt(totalGrossWeightMin.val()) > parseInt(totalGrossWeightMax.val())) {
                alert("Max value must be greater then min value");
                totalGrossWeightMax.val("");

            }
        });
        totalValueMin.change(function () {
            if (parseInt(totalValueMax.val()) < parseInt(totalValueMin.val())) {
                alert("Max value must be greater then min value");
                totalValueMin.val("");
            }
        });

        totalValueMax.change(function () {
            if (parseInt(totalValueMax.val()) < parseInt(totalValueMin.val())) {
                alert("Max value must be greater then min value");
                totalValueMax.val("");

            }
        });
        totalPackagesMin.change(function () {
            if (parseInt(totalPackagesMax.val()) < parseInt(totalPackagesMin.val())) {
                alert("Max value must be greater then min value");
                totalPackagesMin.val("");
            }
        });

        totalPackagesMax.change(function () {
            if (parseInt(totalPackagesMin.val()) > parseInt(totalPackagesMax.val())) {
                alert("Max value must be greater then min value");
                totalPackagesMax.val("");

            }
        });
        totalNetWeightMin.change(function () {
            if (parseInt(totalNetWeightMax.val()) < parseInt(totalNetWeightMin.val())) {
                alert("Max value must be greater then min value");
                totalNetWeightMin.val("");
            }
        });

        totalNetWeightMax.change(function () {
            if (parseInt(totalNetWeightMin.val()) > parseInt(totalNetWeightMax.val())) {
                alert("Max value must be greater then min value");
                totalNetWeightMax.val("");

            }
        });

        var consignmentNumber = $("#ConsignmentNumber");
        var uCR = $("#UCR");
        var lRN = $("#LRN");
        var eADMRN = $("#EADMRN");

        consignmentNumber.change(function () {
            if (consignmentNumber.val() === null || consignmentNumber.val() === "" || typeof (consignmentNumber.val()) === "undefined") {
                DateSetValue();
            } else {
                DateSet();
            }
        });
        uCR.change(function () {
            if (uCR.val() === null || uCR.val() === "" || typeof (uCR.val()) === "undefined") {
                DateSetValue();
            } else {
                DateSet();
            }
        });
        lRN.change(function () {
            if (lRN.val() === null || lRN.val() === "" || typeof (lRN.val()) === "undefined") {
                DateSetValue();
            } else {
                DateSet();
            }
        });
        eADMRN.change(function () {
            if (eADMRN.val() === null || eADMRN.val() === "" || typeof (eADMRN.val()) === "undefined") {
                DateSetValue();
            } else {
                DateSet();
            }
        });
        if (consignmentNumber.val() === null || consignmentNumber.val() === "" || typeof (consignmentNumber.val()) === "undefined") {
            DateSetValue();
        }
        if (uCR.val() === null || uCR.val() === "" || typeof (uCR.val()) === "undefined") {
            DateSetValue();
        }
        if (lRN.val() === null || lRN.val() === "" || typeof (lRN.val()) === "undefined") {
            DateSetValue();
        }
        if (eADMRN.val() === null || eADMRN.val() === "" || typeof (eADMRN.val()) === "undefined") {
            DateSetValue();
        }
        function DateSetValue() {
            dateCons.val(date.val());
            dateMaxCons.val(dateMax.val());
            expectedDateCons.val(expectedDate.val());
            expectedDateMaxCons.val(expectedDateMax.val());

            if (typeof (expectedDateDep.val()) !== "undefined") {
                expectedDateCons.val(expectedDateDep.val());
            }
            if (typeof (expectedDateMaxDep.val()) !== "undefined") {
                expectedDateMaxCons.val(expectedDateMaxDep.val());
            }
            if (typeof (expectedDate.val() !== null || expectedDate.val() !== "" || typeof (expectedDate.val()) !== "undefined") || typeof (expectedDateMax.val() !== null || expectedDateMax.val() !== "" || typeof (expectedDateMax.val()) !== "undefined")) {
                dateCons.val("");
                dateMaxCons.val("");
            }
            if (typeof (expectedDateDep.val() !== null || expectedDateDep.val() !== "" || typeof (expectedDateDep.val()) !== "undefined") || typeof (expectedDateMaxDep.val() !== null || expectedDateMaxDep.val() !== "" || typeof (expectedDateMaxDep.val()) !== "undefined")) {
                dateCons.val("");
                dateMaxCons.val("");
            }
        }
        function DateSet() {
            dateCons.val("");
            dateMaxCons.val("");
            expectedDateCons.val("");
            expectedDateMaxCons.val("");
        }

        dateCons.parent().parent().parent().hide();
        dateMaxCons.parent().parent().parent().hide();
        expectedDateCons.parent().parent().parent().hide();
        expectedDateMaxCons.parent().parent().parent().hide();

    }
}

File: M#/UI/Modules/Consignments/ConsignmentSearchForm.cs
using Domain;
using MSharp;

namespace Modules
{
    class ConsignmentSearchForm : FormModule<Domain.ConsignmentSearch>
    {
        public ConsignmentSearchForm()
        {
            HeaderText("Consignment- Into/Out of UK");

            Field(x => x.ConsignmentNumber);
            Field(x => x.UKTrader).DataSource("await (info.CompanyUser != null ?  info.CompanyUser.Company.GetTraderAssociatedCompanies() :  new Company().GetTraderAssociatedCompanies())").AsAutoComplete();
            Field(x => x.Declarant).DataSource("await (info.CompanyUser != null ?  info.CompanyUser.Company.GetDeclarentAssociatedCompanies() : new Company().GetDeclarentAssociatedCompanies())").AsAutoComplete();

            var weight = Box("Total gross weight", BoxTemplate.HeaderBox);

            Field(x => x.TotalGrossWeightMin)
                .Label("Min")
                .Box(weight);

            Field(x => x.TotalGrossWeightMax)
                .Label("Max")
                .Box(weight);

            Field(x => x.InvoiceNumber);

            var totalValue = Box("Total value", BoxTemplate.HeaderBox);

            Field(x => x.TotalValueMin)
                .Label("Min")
                .Box(totalValue);

            Field(x => x.TotalValueMax)
                .Label("Max")
                .Box(totalValue);


            Field(x => x.CommodityCode);

            Field(x => x.UCR)
                .Label("Declaration Unique Consignment Reference (DUCR)");

            Field(x => x.Partner).DataSource("await (info.CompanyUser != null ?  info.CompanyUser.Company.GetPartnerAssociatedCompanies() : new Company().GetPartnerAssociatedCompanies())").AsAutoComplete();


            var totalPackages = Box("Total packages", BoxTemplate.HeaderBox);

            Field(x => x.TotalPackagesMin)
                .Label("Min")
                .Box(totalPackages);

            Field(x => x.TotalPackagesMax)
                .Label("Max")
                .Box(totalPackages);

            var totalNetweight = Box("Total net weight", BoxTemplate.HeaderBox);

            Field(x => x.TotalNetWeightMin)
                .Label("Min")
                .Box(totalNetweight);

            Field(x => x.TotalNetWeightMax)
                .Label("Max")
                .Box(totalNetweight);


            Field(x => x.InvoiceCurrency).AsAutoComplete();
            Field(x => x.Progress).AsAutoComplete();


            Field(x => x.DateCons).ControlCssClass("text-hidden");
            Field(x => x.DateMaxCons).ControlCssClass("text-hidden");
            Field(x => x.ExpectedDateCons).ControlCssClass("text-hidden");
            Field(x => x.ExpectedDateMaxCons).ControlCssClass("text-hidden");

            AutoSet(x => x.IsNcts).Value("false");
            AutoSet(x => x.IsIntoUk).Value("true");

            AutoSet(x => x.User).Value("Context.Current.User().ExtractUser<User>()");
            Button("Search").Icon(FA.Search)
               .OnClick(x =>
               {
                   x.SaveInDatabase();
                   x.CSharp(@"string url = ConsignmentSearch.CreateCookieHistory(info.Item.ID.ToString());
                             Redirect(url);");
                   x.CloseModal(Refresh.Ajax);
               }).IsDefault(true);

            Button("Cancel").OnClick(x => x.CloseModal());
            Button("Clear").OnClick(x =>
            {
                x.CSharp("await ConsignmentSearch.ClearCookieHistory(isNcts: false, isIntoUk: true );");
                x.Go<Admin.ShipmentPage>();
                x.CloseModal();

            });

            ViewModelProperty<CompanyUser>("CompanyUser").Code("info.CompanyUser = await GetCompanyUser();");

            OnBeforeSave("No search iteams").Code(@"if (info.Item.CheckConsigmentFiled(item))
                    {
                        scope.Complete();
                        await ConsignmentSearch.ClearCookieHistory(isNcts: false, isIntoUk: true );
                        return true;
                    }");


            LoadJavascriptModule("scripts/components/consignment-search-form.js");

        }
    }
}